packages/inventory/src/services/inventory-item.ts

```diff
@@ -97,86 +111,87 @@ export default class InventoryItemService extends TransactionBaseService {
-        metadata: data.metadata,
-        hs_code: data.hs_code,
-        mid_code: data.mid_code,
-        material: data.material,
-        weight: data.weight,
-        length: data.length,
-        height: data.height,
-        width: data.width,
-        requires_shipping: data.requires_shipping,
-      })
+  @InjectEntityManager()
+  async create(
+    data: CreateInventoryItemInput,
+    @MedusaContext() context: SharedContext = {}
+  ): Promise<InventoryItem> {
+    const manager = context.transactionManager!
+    const itemRepository = manager.getRepository(InventoryItem)
 
-      const result = await itemRepository.save(inventoryItem)
+    const inventoryItem = itemRepository.create({
+      sku: data.sku,
+      origin_country: data.origin_country,
+      metadata: data.metadata,
+      hs_code: data.hs_code,
+      mid_code: data.mid_code,
+      material: data.material,
+      weight: data.weight,
+      length: data.length,
+      height: data.height,
+      width: data.width,
+      requires_shipping: data.requires_shipping,
+    })
 
-      await this.eventBusService_
-        .withTransaction(manager)
-        .emit(InventoryItemService.Events.CREATED, {
-          id: result.id,
-        })
+    const result = await itemRepository.save(inventoryItem)
 
-      return result
+    await this.eventBusService_?.emit?.(InventoryItemService.Events.CREATED, {
+      id: result.id,
     })
+
+    return result
   }
 
   /**
    * @param inventoryItemId - The id of the inventory item to update.
    * @param update - The updates to apply to the inventory item.
    * @return The updated inventory item.
    */
+  @InjectEntityManager()
   async update(
     inventoryItemId: string,
     data: Omit<
       DeepPartial<InventoryItem>,
       "id" | "created_at" | "metadata" | "deleted_at"
-    >
+    >,
+    @MedusaContext() context: SharedContext = {}
   ): Promise<InventoryItem> {
-    return await this.atomicPhase_(async (manager) => {
-      const itemRepository = manager.getRepository(InventoryItem)
+    const manager = context.transactionManager!
+    const itemRepository = manager.getRepository(InventoryItem)
 
-      const item = await this.retrieve(inventoryItemId)
+    const item = await this.retrieve(inventoryItemId, undefined, context)
 
-      const shouldUpdate = Object.keys(data).some((key) => {
-        return item[key] !== data[key]
-      })
+    const shouldUpdate = Object.keys(data).some((key) => {
+      return item[key] !== data[key]
+    })
 
-      if (shouldUpdate) {
-        itemRepository.merge(item, data)
-        await itemRepository.save(item)
+    if (shouldUpdate) {
+      itemRepository.merge(item, data)
+      await itemRepository.save(item)
 
-        await this.eventBusService_
-          .withTransaction(manager)
-          .emit(InventoryItemService.Events.UPDATED, {
-            id: item.id,
-          })
-      }
+      await this.eventBusService_?.emit?.(InventoryItemService.Events.UPDATED, {
```

Comment:
question: I suppose it is ok since it occurs at the end of the transaction block. But in case we add some more stuff after we shouldn't pass the transaction?

Reply:
here we follow the rule that modules/core don't share db transactions.
extending the eventBusService to group all messages before emitting them will probably be a needed feature in the future